pkg/controller/template/render.go

```diff
@@ -191,7 +191,7 @@ func generateMachineConfigForRole(config *renderConfig, roleName string, roleDir
 		return vs
 	}
 
-	ignCfg, err := transpileToIgn(keySortVals(files), keySortVals(units))
+	ignCfg, err := transpileToIgn(keySortVals(files), keySortVals(units), config.Admin.SSHKey)
```

Comment:
WDYT
creating a machineconfig object with ssh key setup and labeling it such that both master and worker machineconfigpools pick it up. (ie. add both `...role=worker` and `...role=master` labels to this new object).

Reply:
Making a different section for ssh key items worries me a bit. Why not use the Ignition section and honor it for both initial setup and future changes?

Reply:
good point @ashcrow. Even if there is a seperate ssh key item, the sshkey still needs to be set in Passwd.Users otherwise the MCD won't be able to update it later.

Reply:
I'm not sure I follow the @ashcrow  comment... :innocent: do you mind elaborating a little bit

Reply:
Sure :smile:. `Ignition` has a section that deals with ssh keys. We have work that's waiting to merge that will honor that `Ignition` section for both MCD as a daemon usage as well as `onceFrom` usage in BYO. Why not continue to use this section for all ssh key modification/addition/removal? Or, put another way, why have a second place that ssh key content may live?

Reply:
That's exactly i would expect the ssh keys to live, even if you create a new machineconfig object with ignition that only has the `user` section set and label it correctly. MCC will combine it to single ignition blob that has all.

Reply:
@abhinavdahiya Perfect. I think we're all on the same page on where the content would live. Just to verify, it sounds like you recommend adding `worker` and `master` labels to the MC as well, is that correct?

Reply:
> @abhinavdahiya Perfect. I think we're all on the same page on where the content would live. Just to verify, it sounds like you recommend adding `worker` and `master` labels to the MC as well, is that correct?

Yes to the new standalone MC we create for SSH keys.